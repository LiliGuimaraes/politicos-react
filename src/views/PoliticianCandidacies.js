/*
 * Copyright (c) 2016, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import _ from 'lodash'
import React from 'react'

const CandidacyListItem = ({ candidacy }) => {
  const status = candidacy.elected ? 'Eleito' : 'NÃ£o eleito'
  const round_number = candidacy.election_round.round_number

  let info = ''
  if (candidacy.city && candidacy.state) {
    info = 'em ' + candidacy.city.name + '/' + candidacy.state.siglum
  } else if (candidacy.state) {
    info = 'em ' + candidacy.state.name
  }

  return (
    <div className="candidacy-list-item">
      {status} no {round_number}&deg; turno para{' '}
      {candidacy.political_office.name} {info}
    </div>
  )
}

const CandidacyList = ({ candidacies }) => {
  const candidacyItems = candidacies.map(candidacy => {
    return <CandidacyListItem key={candidacy.id} candidacy={candidacy} />
  })

  return <dd>{candidacyItems}</dd>
}

const CandidacyYear = props => {
  return (
    <dl className="dl-horizontal">
      <dt>{props.year}</dt>
      <CandidacyList candidacies={props.candidacies[props.year]} />
    </dl>
  )
}

const PoliticianCandidacies = props => {
  const candidacies = _.groupBy(props.data, x => {
    return new Date(x.election_round.date).getFullYear()
  })
  const candidacyYears = Object.keys(candidacies).map(year => {
    return <CandidacyYear key={year} year={year} candidacies={candidacies} />
  })

  return (
    <div className="panel panel-info candidacies">
      <div className="panel-heading">
        <h3 className="panel-title">Candidaturas</h3>
      </div>
      <div className="panel-body">{candidacyYears}</div>
    </div>
  )
}

export default PoliticianCandidacies
