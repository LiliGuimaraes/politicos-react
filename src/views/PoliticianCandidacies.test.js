/*
 * Copyright (c) 2017, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import renderer from 'react-test-renderer'

import PoliticianCandidacies from './PoliticianCandidacies'

describe('<PoliticianCandidacies />', () => {
  it('renders without crashing', () => {
    const data = [
      {
        city: {
          name: 'Moju',
        },
        elected: false,
        election_round: {
          date: '2000-10-01',
          round_number: '1',
        },
        id: 1424288,
        political_office: {
          name: 'Vereador',
        },
        state: {
          siglum: 'PA',
        },
      },
    ]

    const component = renderer.create(<PoliticianCandidacies data={data} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
