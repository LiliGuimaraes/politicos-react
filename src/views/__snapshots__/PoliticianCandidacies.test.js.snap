// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PoliticianCandidacies /> renders without crashing 1`] = `
<div
  className="panel panel-info candidacies"
>
  <div
    className="panel-heading"
  >
    <h3
      className="panel-title"
    >
      Candidaturas
    </h3>
  </div>
  <div
    className="panel-body"
  >
    <dl
      className="dl-horizontal"
    >
      <dt>
        2000
      </dt>
      <dd>
        <div
          className="candidacy-list-item"
        >
          Não eleito
           no 
          1
          ° turno para
           
          Vereador
           
          em Moju/PA
        </div>
      </dd>
    </dl>
  </div>
</div>
`;
